examples/pipeline_wavernn/wavernn.py

```diff
@@ -0,0 +1,489 @@
+    )
+    parser.add_argument(
+        "--win-length",
+        default=1100,
+        type=int,
+        help="the length of the STFT window",
+    )
+    parser.add_argument(
+        "--f-min",
+        default=40.,
+        type=float,
+        help="the lowest frequency of the lowest band in a spectrogram",
+    )
+    parser.add_argument(
+        "--n-res-block",
+        default=10,
+        type=int,
+        help="the number of ResBlock in stack",
+    )
+    parser.add_argument(
+        "--n-rnn",
+        default=512,
+        type=int,
+        help="the dimension of RNN layer",
+    )
+    parser.add_argument(
+        "--n-fc",
+        default=512,
+        type=int,
+        help="the dimension of fully connected layer ",
+    )
+    parser.add_argument(
+        "--kernel-size",
+        default=5,
+        type=int,
+        help="the number of kernel size in the first Conv1d layer",
+    )
+    parser.add_argument(
+        "--n-freq",
+        default=80,
+        type=int,
+        help="the number of bins in a spectrogram",
+    )
+    parser.add_argument(
+        "--n-hidden",
+        default=128,
+        type=int,
+        help="the number of hidden dimensions",
+    )
+    parser.add_argument(
+        "--n-output",
+        default=128,
+        type=int,
+        help="the number of output dimensions",
+    )
+    parser.add_argument(
+        "--mode",
+        default="RAW",
+        type=str,
+        help="the type of input waveform in ['RAW', 'MOL']",
+    )
+    parser.add_argument(
+        "--seq-len-factor",
+        default=5,
+        type=int,
+        help="seq_length = hop_length * seq_len_factor, the length of sequence for training",
+    )
+    parser.add_argument(
+        "--test-samples",
+        default=50,
+        type=float,
+        help="the number of files for test",
+    )
+    parser.add_argument(
+        "--file-path",
+        default="/private/home/jimchen90/datasets/LJSpeech-1.1/wavs/",
+        type=str,
+        help="the path of audio files",
+    )
+
+    args = parser.parse_args()
+    return args
+
+
+def save_checkpoint(state, is_best, filename):
+
+    if filename == "":
+        return
+
+    tempfile = filename + ".temp"
+
+    # Remove tempfile in case interuption during the copying from tempfile to filename
+    if os.path.isfile(tempfile):
+        os.remove(tempfile)
+
+    torch.save(state, tempfile)
+    if os.path.isfile(tempfile):
+        os.rename(tempfile, filename)
+    if is_best:
+        shutil.copyfile(filename, "model_best.pth.tar")
+
+    print("Checkpoint: saved", flush=True)
+
+
+# count total parameters in the model
+def count_parameters(model):
+    return sum(p.numel() for p in model.parameters() if p.requires_grad)
+
+
+def train_one_epoch(model, criterion, optimizer, data_loader, device, pbar=None):
+
+    model.train()
+
+    sums = defaultdict(lambda: 0.0)
+
+    for i, (x, m, y) in enumerate(data_loader):
```

Comment:
nit: don't forget to remove i once done debugging :)

Reply:
This part has been updated by `bg_iterator`.